addons/binding/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/internal/discovery/Enigma2DiscoveryParticipant.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * @author Thomas Traunbauer - Initial contribution
+ */
+public class Enigma2DiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private Logger logger = LoggerFactory.getLogger(Enigma2DiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Enigma2BindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public DiscoveryResult createResult(ServiceInfo info) {
+        DiscoveryResult result = null;
+        ThingUID uid = getThingUID(info);
+        if (uid != null) {
+            Map<String, Object> properties = new HashMap<>(4);
+            String label = info.getName();
+
+            properties.put(Enigma2BindingConstants.DEVICE_PARAMETER_HOST, getIPAddress(info));
+            properties.put(Enigma2BindingConstants.DEVICE_PARAMETER_REFRESH, new BigDecimal(5));
+            return DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(label).build();
+        }
+        return result;
+    }
+
+    @Override
+    public ThingUID getThingUID(ServiceInfo info) {
+        logger.debug("ServiceInfo: {}", info);
+        if (info != null && getServiceType().equals(info.getType())) {
+            logger.trace("Discovered a Enigma2 device thing with name '{}'", info.getName());
+            String formatedIP = getFormattedIPAddress(info);
+            if (formatedIP != null) {
+                return new ThingUID(Enigma2BindingConstants.THING_TYPE_DEVICE, new String(formatedIP));
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String getServiceType() {
+        // I know, this is not optimal, but currently i have no better idea!
+        return "_ssh._tcp.local.";
+    }
+
+    private boolean isEnigma2Device(String ip) {
+        String content;
+        try {
+            content = HttpUtil.executeUrl("GET", "http://" + ip + "/web/about", 5000);
+        } catch (IOException e) {
+            return false;
+        }
+        return content != null && content.contains("e2enigmaversion");
+    }
+
+    private String getIPAddress(ServiceInfo info) {
+        if (info != null) {
+            InetAddress[] addrs = info.getInetAddresses();
+            List<Inet4Address> listOfInt4Addresses = new ArrayList<>();
+
+            for (int i = 0; i < addrs.length; i++) {
+                if (addrs[i] instanceof Inet4Address) {
+                    listOfInt4Addresses.add((Inet4Address) addrs[i]);
+                }
+            }
+
+            if (listOfInt4Addresses.size() > 1) {
+                logger.info("Enigma2 device {} reports multiple addresses - using the first one! {}", info.getName(),
+                        Arrays.toString(listOfInt4Addresses.toArray(new Inet4Address[listOfInt4Addresses.size()])));
+            }
+
+            if (listOfInt4Addresses.size() > 0) {
+                String ip = listOfInt4Addresses.get(0).getHostAddress();
+                if (isEnigma2Device(ip)) {
```

Comment:
Ouch - you should not do an external HTTP call here. Is there no other way to figure out from the ServiceInfo if it is the expected device?